File: rielarmoryspringboot/src/main/java/com/gjbmloslos/rielarmoryspringboot/service/RoleService.java
package com.gjbmloslos.rielarmoryspringboot.service;

import com.gjbmloslos.rielarmoryspringboot.model.Role;

import java.util.List;

public interface RoleService {
    Role getRoleById(Long id);
    List<Role> getAllRoles();
}

File: rielarmoryangular/src/app/service/role.service.ts
import { Injectable } from '@angular/core';
import { Role } from '../model/role.model';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class RoleService {

  private apiUrl = '/api/roles';

  constructor(private http: HttpClient) { }

  getAllRoles(): Observable<Role[]> {
    return this.http.get<Role[]>(`${this.apiUrl}/all`);
  } 
}
